packages/python/plotly/plotly/express/_core.py

```diff
@@ -2280,19 +2512,25 @@ def make_figure(args, constructor, trace_patch=None, layout_patch=None):
                 base = args["x"] if args["orientation"] == "v" else args["y"]
                 var = args["x"] if args["orientation"] == "h" else args["y"]
                 ascending = args.get("ecdfmode", "standard") != "reversed"
-                group = group.sort_values(by=base, ascending=ascending)
-                group_sum = group[var].sum()  # compute here before next line mutates
-                group[var] = group[var].cumsum()
+                group = group.sort(by=base, descending=not ascending, nulls_last=True)
+                group_sum = group.get_column(
+                    var
+                ).sum()  # compute here before next line mutates
+                group = group.with_columns(**{var: nw.col(var).cum_sum()})
                 if not ascending:
-                    group = group.sort_values(by=base, ascending=True)
+                    group = group.sort(by=base, descending=False, nulls_last=True)
 
                 if args.get("ecdfmode", "standard") == "complementary":
-                    group[var] = group_sum - group[var]
+                    group = group.with_columns(
+                        **{var: -nw.col(var) + nw.lit(group_sum)}
+                    )
```

Comment:
I don't fully understand this section, but I was trying building the docs pages locally and noted [this example](https://plotly.com/python/ecdf-plots/#reversed-and-complementary-cdf-plots) throws an error here:

```
import plotly.express as px
fig = px.ecdf(df, x=[1,2,3,4], markers=True, ecdfmode="complementary",
              title="ecdfmode='complementary' (Y=fraction above X value)")
fig.show()
```

```
File ~/Desktop/nw-tests/venv/lib/python3.12/site-packages/plotly/express/_core.py:2525, in make_figure(args, constructor, trace_patch, layout_patch)
   2521     group = group.sort(by=base, descending=False, nulls_last=True)
   2523 if args.get("ecdfmode", "standard") == "complementary":
   2524     group = group.with_columns(
-> 2525         **{var: -nw.col(var) + nw.lit(group_sum)}
   2526     )
   2528 if args["ecdfnorm"] == "probability":
   2529     group = group.with_columns(**{var: nw.col(var) / nw.lit(group_sum)})

TypeError: bad operand type for unary -: 'Expr'
```

I am using:
```
narwhals                      1.9.2
pandas                        2.2.3
```

Reply:
Thanks Liam, I will take a look! Reason for `-nw.col(var) + nw.lit(group_sum)` is to avoid falling into the issue with left most `lit` operand ([narwhals#853](https://github.com/narwhals-dev/narwhals/issues/853)).

I would expecte the solution to be something like: `(nw.col(var) - nw.lit(group_sum))*(-1)`